### PR TITLE
Fix behavior for Unknown states

### PR DESCRIPTION
**Pull request by kcover**
#### What does this PR do?
Fixes some behavior with unknown states.

#### How should this be tested? (List steps with links to updated documentation)
CI

#### Checklist:
- [ ] Documentation Updated
- [x] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

File: commons-api/src/main/java/com/connexta/transformation/commons/api/status/Transformation.java [hunk 1]
```diff
@@ -76,6 +76,7 @@ default boolean wasSuccessful() {
 
   @Override
   default boolean isUnknown() {
-    return isCompleted() && metadatas().anyMatch(MetadataTransformation::isUnknown);
+    return metadatas().anyMatch(MetadataTransformation::isUnknown)
+        && metadatas().noneMatch(metadataTransformation -> getState() == State.IN_PROGRESS);
```
Comment (paouelle) [inline commons-api/src/main/java/com/connexta/transformation/commons/api/status/Transformation.java]:
```suggestion
        && metadatas().map(MetadataTransformation::getState).noneMatch(State.IN_PROGRESS);
```

or better ...

```suggestion
        && metadatas().noneMatch(MetadataTransformation ::isInProgress);
```

File: commons-api/src/main/java/com/connexta/transformation/commons/api/status/TransformationStatus.java [hunk 2]
```diff
@@ -77,7 +77,7 @@
   default boolean isCompleted() {
     final State state = getState();
 
-    return state != State.IN_PROGRESS;
+    return state == State.SUCCESSFUL || state == State.FAILED;
```
Comment (paouelle) [inline commons-api/src/main/java/com/connexta/transformation/commons/api/status/TransformationStatus.java:80]: Nice catch. Hope you added test cases 😄

File: commons-api/src/test/java/com/connexta/transformation/commons/api/status/TransformationStatusTest.java [hunk 3]
```diff
@@ -43,6 +43,12 @@ public void isCompletedInProgress() {
     assertFalse(status.isCompleted());
   }
 
+  @Test
+  public void isCompletedUnknown() {
```
Comment (paouelle) [inline commons-api/src/test/java/com/connexta/transformation/commons/api/status/TransformationStatusTest.java:47]: 🎉 Nice ... see the importance of testing all enums possibilities

**Review by paouelle — APPROVED** [review]

File: commons-api/src/test/java/com/connexta/transformation/commons/api/status/TransformationStatusTest.java (hunk 3 above)
Comment (tyler30clemens) [reply commons-api/src/test/java/com/connexta/transformation/commons/api/status/TransformationStatusTest.java:47]: Would it make more sense to name this `hasNotCompletedWithUnknown`? Same with the test above to be called `hasNotCompletedWithInProgress`?

**Review by tyler30clemens — APPROVED** [review]

**Review by brjeter — APPROVED** [review]
I approve, but I've noticed a separate issue with statuses. The `Transformation` class overrides `getState()` in order to aggregate and reduce the metadatas' statuses, but then the status helper functions like `isCompleted()` basically duplicate the aggregation logic
https://github.com/connexta/ion-transformation-storage/blob/372c204fac0a6c7778d79b3f7f0dd288ec289297/commons-api/src/main/java/com/connexta/transformation/commons/api/status/Transformation.java#L62-L65

when they could just use the default implementation along with the aggregating `getState()` method
https://github.com/connexta/ion-transformation-storage/blob/372c204fac0a6c7778d79b3f7f0dd288ec289297/commons-api/src/main/java/com/connexta/transformation/commons/api/status/TransformationStatus.java#L77-L81

File: commons-api/src/main/java/com/connexta/transformation/commons/api/status/TransformationStatus.java (hunk 2 above)
Comment (kcover) [reply commons-api/src/main/java/com/connexta/transformation/commons/api/status/TransformationStatus.java:80]: I did. 🖊

**Pull request merged**